service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/ControlledAzureManagedIdentityResource.java

```diff
@@ -0,0 +1,228 @@
+    RetryRule cloudRetry = RetryRules.cloud();
+    flight.addStep(
+        new GetAzureManagedIdentityStep(
+            flightBeanBag.getAzureConfig(), flightBeanBag.getCrlService(), this),
+        cloudRetry);
+    flight.addStep(
+        new CreateAzureManagedIdentityStep(
+            flightBeanBag.getAzureConfig(), flightBeanBag.getCrlService(), this),
+        cloudRetry);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void addDeleteSteps(DeleteControlledResourcesFlight flight, FlightBeanBag flightBeanBag) {
+    flight.addStep(
+        new DeleteAzureManagedIdentityStep(
+            flightBeanBag.getAzureConfig(), flightBeanBag.getCrlService(), this),
+        RetryRules.cloud());
+  }
+
+  // Azure resources currently do not implement updating.
+  @Override
+  public void addUpdateSteps(UpdateResourceFlight flight, FlightBeanBag flightBeanBag) {}
+
+  public ApiAzureManagedIdentityResource toApiResource() {
+    return new ApiAzureManagedIdentityResource()
+        .metadata(super.toApiMetadata())
+        .attributes(toApiAttributes());
+  }
+
+  private ApiAzureManagedIdentityAttributes toApiAttributes() {
+    return new ApiAzureManagedIdentityAttributes().managedIdentityName(getManagedIdentityName());
+  }
+
+  @Override
+  public String attributesToJson() {
+    return DbSerDes.toJson(
+        new ControlledAzureManagedIdentityAttributes(getManagedIdentityName(), getRegion()));
+  }
+
+  @Override
+  public ApiResourceAttributesUnion toApiAttributesUnion() {
+    ApiResourceAttributesUnion union = new ApiResourceAttributesUnion();
+    union.azureManagedIdentity(toApiAttributes());
+    return union;
+  }
+
+  @Override
+  public void validate() {
+    super.validate();
+    if (getResourceType() != WsmResourceType.CONTROLLED_AZURE_MANAGED_IDENTITY
+        || getResourceFamily() != WsmResourceFamily.AZURE_MANAGED_IDENTITY
+        || getStewardshipType() != StewardshipType.CONTROLLED) {
+      throw new InconsistentFieldsException("Expected controlled AZURE_MANAGED_IDENTITY");
+    }
+    if (getManagedIdentityName() == null) {
+      throw new MissingRequiredFieldException(
+          "Missing required managedIdentityName field for ControlledAzureManagedIdentity.");
+    }
+    if (getRegion() == null) {
+      throw new MissingRequiredFieldException(
+          "Missing required region field for ControlledAzureManagedIdentity.");
+    }
+    AzureResourceValidationUtils.validateAzureManagedIdentityName(getManagedIdentityName());
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    if (!super.equals(o)) return false;
+
+    ControlledAzureManagedIdentityResource that = (ControlledAzureManagedIdentityResource) o;
+
+    return managedIdentityName.equals(that.getManagedIdentityName());
+  }
+
+  @Override
+  public boolean partialEqual(Object o) {
+    if (this == o) return true;
```

Comment:
This implementation looks identical to `equals`. If they should have the same implementation, any reason to not just call equals from here? (I don't know why there is a `partialEqual` method at all)